src/server.ts

```diff
@@ -44,17 +45,12 @@ async function handleRequest(
   root: string,
   options: WebServerOptions
 ) {
-  const pathParts = req.url?.split('/') || [];
+  const pathParts = new URL(req.url || '', 'http://abc').pathname.split('/');
   const originalPath = path.join(root, ...pathParts);
   if (req.url?.endsWith('/')) {
```

Comment:
This would be a bug too, right?  This is looking for a route like `/test/path/` and turning it into `/test/path/index.html`.  If it were `/test/path/?a=b`, I think we'd want `/test/path/index.html`.  I would stache the result of the url:

```js
const url = new URL(req.url || '/', `http://localhost:${options.port}`);
const pathParts = url.pathname.split('/') || [];
```

And then use `url.pathname` to check:

```js
if (url.pathname.endsWith('/')) {
```